Loadtree

albert-smit opened the pull request:
the Phylogenetic tree is now also loaded from a file, and opened from the open dialog

albert-smit committed:
load a newick file

albert-smit committed:
added filename determination for newick files

albert-smit committed:
open dialog now opens .nwk files too

albert-smit committed:
newick files are now opened with an absolute path

albert-smit committed:
fixed checkstyle warnings

jorenham commented on src/main/java/nl/tudelft/lifetiles/core/controller/MenuController.java:
```diff
@@ -132,6 +137,32 @@ private String collectGraphFileName(final File directory)
+     *            the directory in which the .nwk file is located
+     * @return the name of the nwk file without its extension
+     * @throws IOException
+     *             throws an exception when no, or too many files have been
```
Iffy indentation

jorenham commented on src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java:
```diff
@@ -169,6 +187,37 @@ public boolean isLoaded() {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        String fileString = sc.next();
```
Beware the potential NullPointer

jorenham commented on src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java:
```diff
@@ -169,6 +187,37 @@ public boolean isLoaded() {
+            e.printStackTrace();
+        }
+        String fileString = sc.next();
+        PhylogeneticTreeFactory np = new PhylogeneticTreeFactory(fileString);
```
The factory ought to be passed to the method

albert-smit committed:
fixed comments
no more potential null pointer in loadTree()
no more weird indentation
LoadTree now takes a File

albert-smit committed:
changed PhylogeneticTreeFactory to PhylogeneticTreeParser
because it never was a factory

albert-smit committed:
changed assert statements in MenuController to ifs

jorenham commented on src/main/java/nl/tudelft/lifetiles/core/controller/ViewController.java:
```diff
@@ -176,6 +193,37 @@ public boolean isLoaded() {
+        }
+
+        //parse the string into a tree
+        PhylogeneticTreeParser np = new PhylogeneticTreeParser(fileString);
```
This is not very DI, you should use a factory or make it static.

albert-smit committed:
made PhylogeneticTreeParser.parse static, and the class uninstantiable
